octodns/provider/constellix.py

```diff
@@ -194,12 +238,41 @@ def __init__(self, id, api_key, secret_key, ratelimit_delay=0.0,
 
     def _data_for_multiple(self, _type, records):
         record = records[0]
+        if record['recordOption'] == 'pools':
+            return self._data_for_pool(_type, record)
         return {
             'ttl': record['ttl'],
             'type': _type,
             'values': record['value']
         }
 
+    def _data_for_pool(self, _type, record):
+        pool_id = record['pools'][0]
+        pool = self._client.pool_by_id(_type, pool_id)
+        pool_name = pool['name'].split(':')[-1]
+        pools = {}
+        values = []
+        pools[pool_name] = {
+            'values': []
+        }
+        for value in pool['values']:
+            pools[pool_name]['values'].append({
+                'value': value['value'],
+                'weight': value['weight']
+            })
+            values.append(value['value'])
+        return {
+            'ttl': record['ttl'],
+            'type': _type,
+            'dynamic': {
+                'pools': pools,
+                'rules': [{
+                    'pool': pool_name
+                }]
```

Comment:
👍 This is a good way to incrementally work towards dynamic records. It should plan & apply fine for anything with a single pool and only a default rule.